clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -4185,9 +4188,16 @@ void CodeGenModule::emitMultiVersionFunctions() {
       continue;
     }
 
-    if (!EmitResolver)
+    if (!ShouldEmitResolver)
       continue;
 
+    if (!HasDefaultDecl) {
+      FunctionDecl *NewFD = createDefaultTargetVersionFrom(FD);
+      llvm::Constant *Func = createFunction(NewFD);
```

Comment:
nit: the cast would be a tiny bit more useful up here.
```suggestion
      auto *Func = cast<llvm::Function>(createFunction(NewFD));
```